mkat_fpga_tests/test_cbf.py

```diff
@@ -583,4 +585,96 @@ def test_sensor_values(self):
         """
         (TP.C.1.16) Report sensor values (AR1)
         """
-        pass
+        def cmc_sensor_status():
+            """ Check cmc sensor status """
+            iom = ioloop_manager.IOLoopManager()
+            iow = resource_client.IOLoopThreadWrapper(iom.get_ioloop())
+            iom.start()
+            self.addCleanup(iom.stop)
+            #roach_clients = []
+            rc = resource_client.KATCPClientResource(
+                dict(name='localhost', address=('localhost', '7147'),
+                    controlled=True))
+            rc.set_ioloop(iom.get_ioloop())
+            rct = resource_client.ThreadSafeKATCPClientResourceWrapper(rc, iow)
+            rct.start()
+            rct.until_synced()
+            #roach_clients.append(rct)
+
+            ## 1. Request a list of available sensors using KATCP command
+            ## 2. Confirm the CBF replies with a number of sensor-list inform messages
+            LOGGER.info (rct.req.sensor_list())
+
+            # 3. Confirm the CBF replies with "!sensor-list ok numSensors"
+            #   where numSensors is the number of sensor-list informs sent.
+            sens_val_stat, sens_val_cnt =  rct.req.sensor_value().reply.arguments
+            sens_lst_stat, sens_lst_cnt = rct.req.sensor_list().reply.arguments
+
+            self.assertTrue(sens_val_cnt == sens_lst_cnt,
+                msg='Sensors count are not the same')
+
+            # 3. Confirm the CBF replies with "!sensor-list ok numSensors" where
+            # numSensors is the number of sensor-list informs sent.
+            #print (str(rct.req.sensor_value().reply))
+            LOGGER.info (str(rct.req.sensor_value().reply))
+
+            # Sensors status
+            rct.sensor.time_synchronised.status
```

Comment:
You must read the sensor value before the sensor object's status will be correct. I recommend calling rct.sensor.time_synchronised.get_reading(), which will cause the sensor value to be read via KATCP and should return a structure with value, status and update time.